website/api/recursion.md

```diff
@@ -0,0 +1,122 @@
+j2-->suc12("<a href='https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.SuccinctReceipt.html'>SuccinctReceipt</a>")
+j3-->suc13("<a href='https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.SuccinctReceipt.html'>SuccinctReceipt</a>")
+
+suc11-->j11("<a href='https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.ApiClient.html#method.join'>.join()</a>")
+suc12-->j11
+
+j11-->suc21("<a href='https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.SuccinctReceipt.html'>SuccinctReceipt</a>")
+
+suc21-->j21("<a href='https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.ApiClient.html#method.join'>.join()</a>")
+suc13-->j21
+
+j21-->suc31("<a href='https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.SuccinctReceipt.html'>SuccinctReceipt</a>")
+
+suc31-->id("<a href='https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.ApiClient.html#method.identity_p254'>.identity_p254()</a>")
+
+id-->suc41("<a href='https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.SuccinctReceipt.html'>SuccinctReceipt</a>")
+
+suc41-->compress("<a href='https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.ApiClient.html#method.compress'>.compress()</a>")
+
+compress-->groth16("<a href='https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.Groth16Receipt.html'>Groth16Receipt</a>")
+```
+
+To summarize the diagram:
+1. The program is *executed*, resulting in a collection of *Segments*.
+1. Each *Segment* is proven, resulting in a *SegmentReceipt*.
+1. Each *SegmentReceipt* is *lifted*, resulting in a *SuccinctReceipt*.
+1. Pairs of *SuccinctReceipts* are *joined*, resulting in another *SuccinctReceipt*. This continues until a single *SuccinctReceipt* remains.
+1. The final *SuccinctReceipt* is passed through *identity_p254*, which prepares for Groth16 proving.
+1. The *SuccinctReceipt* is *compressed*, generating a *Groth16Receipt*.
+
+The *Groth16Receipt* can now be posted on-chain and verified by the [RISC Zero Verifier Contract].
+
+## Recursive Circuit Architecture
+
+RISC Zero's zkVM consists of three circuits.
+
+1. The RISC-V Circuit is a STARK circuit that proves correct execution of RISC-V programs.
+1. The Recursion Circuit is a separate STARK circuit, that's designed to efficiently generate proofs for the verification of STARK proofs and to support the integration of custom accelerator circuits into the zkVM.
+This circuit has a similar architecture to the RISC-V Circuit, but with fewer columns and a much simpler instruction set.
+The Recursion Circuit supports a number of programs, including [lift], [join], [resolve], and [identity_p254].
+The same [proof system] is used for both the RISC-V Circuit and the Recursion Circuit.
+1. The STARK-to-SNARK Circuit is an R1CS circuit that proves verification of proofs from the *identity_p254* program.
+
+*The size of a [Receipt] varies depending on the size of the ReceiptClaim and its ReceiptKind. For SuccinctReceipts and Groth16 Receipts, the size of the *seal* is constant.
```

Comment:
I don't think you need this caveat. The proof itself is indeed constant size, and this is true of any ZKP, even as an ideal block box.